pkg/scheduler/scheduler.go

```diff
@@ -139,36 +197,77 @@ func New(client clientset.Interface,
 	for _, opt := range opts {
 		opt(&options)
 	}
-	// Set up the configurator which can create schedulers from configs.
-	configurator := factory.NewConfigFactory(&factory.ConfigFactoryArgs{
-		SchedulerName:                  options.schedulerName,
-		Client:                         client,
-		NodeInformer:                   nodeInformer,
-		PodInformer:                    podInformer,
-		PvInformer:                     pvInformer,
-		PvcInformer:                    pvcInformer,
-		ReplicationControllerInformer:  replicationControllerInformer,
-		ReplicaSetInformer:             replicaSetInformer,
-		StatefulSetInformer:            statefulSetInformer,
-		ServiceInformer:                serviceInformer,
-		PdbInformer:                    pdbInformer,
-		StorageClassInformer:           storageClassInformer,
+	stopEverything := stopCh
+	if stopEverything == nil {
+		stopEverything = wait.NeverStop
+	}
+	schedulerCache := internalcache.New(30*time.Second, stopEverything)
+
+	// TODO(bsalamat): config files should be passed to the framework.
+	framework, err := framework.NewFramework(registry, nil)
+	if err != nil {
+		klog.Fatalf("error initializing the scheduling framework: %v", err)
+	}
+	// storageClassInformer is only enabled through VolumeScheduling feature gate
+	var storageClassLister storagelisters.StorageClassLister
+	if storageClassInformer != nil {
+		storageClassLister = storageClassInformer.Lister()
+	}
+
+	podQueue := internalqueue.NewSchedulingQueue(stopEverything)
+	// Setup volume binder
+	volumeBinder := volumebinder.NewVolumeBinder(client, nodeInformer, pvcInformer, pvInformer, storageClassInformer, time.Duration(options.bindTimeoutSeconds)*time.Second)
+	scheduledPodsHasSynced := podInformer.Informer().HasSynced
+	// ScheduledPodLister is something we provide to plug-in functions that
+	// they may need to call.
+	// scheduledPodLister := assignedPodLister{podInformer.Lister()}
```

Comment:
Can this be deleted? I believe this is referring to the old concept of factory plugins and not the new framework plugins.